components/TaskCard.vue

```diff
@@ -70,23 +93,85 @@ export default {
 
   created() {
     this.task = this.cardData;
+    this.taskDescription = this.task.description;
   },
   methods: {
     async markDone() {
-      this.loading = true;
-      await this.$store.dispatch('todos/changeTaskState', this.task);
+      if (!this.showEditIcon) {
+        if (this.taskDescription.length <= 0) {
+          this.titleInputError = true;
+          this.titleErrorMsg = 'Field is empty';
+
+          return;
+        }
+
+        this.loading = true;
+        const val = {
+          description: this.taskDescription,
+          id: this.task.id,
+        };
+
+        await this.$store.dispatch('todos/editTask', val);
+        await this.$store.dispatch('todos/changeTaskState', this.task);
+        this.titleInputError = false;
+        this.titleErrorMsg = '';
+
+        if (this.requestInProcess) return;
 
-      if (!this.requestInProcess) {
         this.loading = false;
+        this.showEditIcon = true;
+
+        return;
```

Comment:
If I understand correctly, the above block updates and marks a todo as done if the mark done button is clicked during edit. note that we are doing the similar things seperately inside `editTask()` method and below in this function. Can we rearrange this function so that same code isn't manually added twice?

Reply:
Resolved